modules/tftp/server.rb

```diff
@@ -93,7 +93,13 @@ def pxeconfig_file mac
   end
 
   def self.fetch_boot_file dst, src
-    filename    = dst + '-' + src.split("/")[-1]
+    # If dst file name contains a slash - we don't use a hyphen (file goes into a subdirectory)
+    if dst.include? "/"      
+      filename  = dst + src.split("/")[-1]
+    else
+      filename  = dst + '-' + src.split("/")[-1]
+    end
+
```

Comment:
I'm afraid this is still not going to work: on line 109 https://github.com/kireevco/smart-proxy/blob/develop/modules/tftp/server.rb#L109, the resulting directory tree won't have the last directory, if I correctly understand the directory layout suggested in #9299. 

Is the directory per architecture required for pxe booting of Windows? Could we get away with our current approach, i.e. prefixing boot files with the os and architecture name? For example, the names of boot files for centos 6.4 for x86 are: CentOS-6.4-x86_64-initrd.img and CentOS-6.4-x86_64-vmlinuz.